### PR TITLE
BUG: Keep itk.Image reference in NumPy array views

### PR DESCRIPTION
**Pull request by thewtex**
Prevent the view'ed image from going out of scope by holding a reference
in a new class, NDArrayITKBase. The object is kept in the `itk_base`
attributed -- the typical `base` attribute NumPy uses for this purpose
is not writable. The class is created according to the documentation
here:

  https://numpy.org/doc/stable/user/basics.subclassing.html

### CODE REVIEW

File: Modules/Bridge/NumPy/wrapping/PyBuffer.i.in [hunk 1]
```diff
@@ -77,14 +74,10 @@
         C-order indexing, i.e. k,j,i, the image Size will have the dimensions
         reversed from the array shape.
 
-        Therefore, since the *numpy.transpose* operator on a 2D array simply
+        Therefore, since the *np.transpose* operator on a 2D array simply
```
Comment (dzenanz) [inline Modules/Bridge/NumPy/wrapping/PyBuffer.i.in:77]: Was this change intentional? `numpy` is more descriptive than `np`.

File: Modules/Bridge/NumPy/wrapping/PyBuffer.i.init [hunk 2]
```diff
@@ -2,7 +2,20 @@
 
 HAVE_NUMPY = True
 try:
-  import numpy
+  import numpy as np
+  class NDArrayITKBase(np.ndarray):
+      """A numpy array with an optional itk object "base" object to used to
```
Comment (dzenanz) [inline Modules/Bridge/NumPy/wrapping/PyBuffer.i.init:7]: clean up sentence

File: Modules/Bridge/NumPy/wrapping/PyBuffer.i.in (hunk 1 above)
Comment (thewtex) [reply Modules/Bridge/NumPy/wrapping/PyBuffer.i.in:77]: Yes, also update this, but I missed one spot -- adding a  follow-up. In the Python community `np` is standard practice.

File: Modules/Bridge/NumPy/wrapping/PyBuffer.i.init (hunk 2 above)
Comment (thewtex) [reply Modules/Bridge/NumPy/wrapping/PyBuffer.i.init:7]: Good catch.

Much improved now: *A numpy array that provides a view on the data associated with an option itk "base" object.*
